File: Backend/app/schemas/lifecycle.py
"""Lifecycle / NBA API schemas."""
from typing import Any, Dict, List, Optional
from pydantic import BaseModel


class ProductConsumption(BaseModel):
    product_id: str
    name: str
    purchased: bool
    deployed: bool
    current_pct: int
    target_pct: int


class LifecycleAlertResponse(BaseModel):
    account_id: str
    account_name: Optional[str] = None
    stage: str
    stage_label: str
    priority: int
    health_status: str
    score: int
    why_now: str
    consumption: List[ProductConsumption]
    contract_month: int
    contract_total_months: int
    recommended_channel: str
    recommended_channel_label: str
    recommended_action: str
    channel_reason: str
    due_hint: str
    history_insights: List[str] = []


class NextBestActionResponse(BaseModel):
    id: str
    account_id: str
    account_name: Optional[str] = None
    stage: str
    stage_label: str
    action: str
    priority: int
    due_hint: str


class AgentRecommendationResponse(BaseModel):
    actions: List[str]
    assets: List[Dict[str, str]]
    data_insight: str


class PortfolioConsumptionResponse(BaseModel):
    avg_deployment_pct: int
    products_at_risk: int
    unused_entitlements: int
    accounts_needing_action: int


class LifecycleDashboardResponse(BaseModel):
    stage_alerts: List[LifecycleAlertResponse]
    account_alerts: List[LifecycleAlertResponse]
    nba_items: List[NextBestActionResponse]
    selected_alert: Optional[LifecycleAlertResponse] = None
    agent_recommendation: Optional[AgentRecommendationResponse] = None
    portfolio_consumption: PortfolioConsumptionResponse
    stage_counts: Dict[str, int]


class ActionRecommendationResponse(BaseModel):
    channel: str
    channel_label: str
    action: str
    why_now: str
    channel_reason: str
    stage: str
    stage_label: str
    due_hint: str
    history_insights: List[str] = []

File: revenue-navigator/src/types/lifecycle.ts
import type { Account } from "@/data/mockData";

export type LifecycleStageId = "protect" | "renew" | "adopt" | "expand" | "activate";

export type HealthStatus = "At Risk" | "Watch" | "Good" | "Excellent" | "New Account";

export interface ProductConsumption {
  productId: string;
  name: string;
  purchased: boolean;
  deployed: boolean;
  currentPct: number;
  targetPct: number;
}

export interface LifecycleStageConfig {
  id: LifecycleStageId;
  label: string;
  priority: number;
  accentClass: string;
  borderClass: string;
  badgeClass: string;
}

export interface LifecycleAlert {
  account: Account;
  stage: LifecycleStageId;
  stageLabel: string;
  priority: number;
  healthStatus: HealthStatus;
  score: number;
  whyNow: string;
  consumption: ProductConsumption[];
  contractMonth?: number;
  contractTotalMonths?: number;
}

export interface NextBestAction {
  id: string;
  accountId: string;
  accountName: string;
  stage: LifecycleStageId;
  stageLabel: string;
  action: string;
  priority: number;
  dueHint?: string;
}

export interface AgentRecommendation {
  actions: string[];
  assets: { label: string; type: "guide" | "kit" | "template" }[];
  dataInsight: string;
}

export interface LifecycleDashboardData {
  stageAlerts: LifecycleAlert[];
  nbaItems: NextBestAction[];
  selectedAlert: LifecycleAlert | null;
  agentRecommendation: AgentRecommendation | null;
  portfolioConsumption: {
    avgDeploymentPct: number;
    productsAtRisk: number;
    unusedEntitlements: number;
    accountsNeedingAction: number;
  };
}

export type RecommendedChannel = "call" | "message" | "email";

export interface ActionRecommendation {
  channel: RecommendedChannel;
  channelLabel: string;
  action: string;
  whyNow: string;
  channelReason: string;
  stage: LifecycleStageId;
  stageLabel: string;
  dueHint: string;
}
